src/cbc_sdk/workload/sensor_lifecycle.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+
+# *******************************************************
+# Copyright (c) VMware, Inc. 2020. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Sensor Lifecycle Management for Workloads"""
+
+from cbc_sdk.errors import ApiError
+from cbc_sdk.base import (UnrefreshableModel, BaseQuery, CriteriaBuilderSupportMixin,
+                          IterableQueryMixin, AsyncQueryMixin)
+
+import logging
+import json
+
+log = logging.getLogger(__name__)
+
+
+class SensorKit(UnrefreshableModel):
+    """Represents the information about a sensor, including installation file URLs."""
+    swagger_meta_file = "workload/models/sensorKit.yaml"
+
+    VALID_DEVICE_TYPES = ["WINDOWS", "LINUX", "MAC"]
+    VALID_ARCHITECTURES = ["32", "64", "OTHER"]
+    VALID_TYPES = ["WINDOWS", "MAC", "RHEL", "UBUNTU", "SUSE", "AMAZON_LINUX"]
+
+    def __init__(self, cb, initial_data=None):
+        """
+        Initialize the SensorKit object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            initial_data (dict): Initial data used to populate the sensor kit data.
+        """
+        super(SensorKit, self).__init__(cb, None, initial_data)
+        self._full_init = True
```

Comment:
This should not be considered _full_init unless the sensor kit url has been fetched i.e. `sensor_url` otherwise it's only a partial init

Reply:
I'm putting in a workaround for this.  Let me know what you think when you see it.

Reply:
That seems reasonable. Is there any chance that a model marked as pseudo would be updated or fetch the sensorkit and therefore making it a full initialized model?

Reply:
I don't see how.  SensorKit is defined as an UnrefreshableModel, so you can't refresh it.  I don't think you can save it either, since there's no save mechanism defined.

Reply:
Cool! Thanks for verifying